lib/kafka_consumer/server.ex

```diff
@@ -44,18 +44,35 @@ defmodule KafkaConsumer.Server do
   Consume messages from topic or stop consumer if topic not found
   """
   @spec consume(Map.t, pid) :: atom
-  def consume(%{topic: topic, partition: partition, worker_name: worker_name,
-      handler: handler, handler_pool: handler_pool}, consumer) do
+  def consume(%{topic: topic, partition: partition} = consume_map, consumer) do
     if Utils.topic_exists?(topic, partition) do
-      Utils.prepare_stream(worker_name)
-      for message <- KafkaEx.stream(topic, partition, worker_name: worker_name) do
-        :poolboy.transaction(handler_pool, fn(pid) ->
-          handler.handle_event(pid, {topic, partition, message})
-        end)
-      end
+      stream(consume_map)
     else
       send consumer, :topic_not_found
     end
     :ok
   end
+
+  defp stream(%{worker_name: worker_name, topic: topic,
+  partition: partition, handler_pool: handler_pool, handler: handler}) do
+    Utils.prepare_stream(worker_name)
+    for message <- KafkaEx.stream(topic, partition, consumer_options(worker_name, topic, partition)) do
+      :poolboy.transaction(handler_pool, fn(pid) ->
+        handler.handle_event(pid, {topic, partition, message})
+      end)
+    end
+  end
+
+  defp consumer_options(worker_name, topic, partition) do
+    case Utils.has_consumer_group? do
+      false ->
+        [
+          worker_name: worker_name,
+          auto_commit: false,
+          offset: Utils.offset_server().get_latest_offset(topic, partition)
+        ]
```

Comment:
It's wrong because you should get latest offset from OffsetServer otherwise you lose messages.
Example:
1) You start consumer, latest offset = 5
2) Producer send two messages. Consumer consume it. latest offset = 7
3) Consumer is down.
4) Producer send two messages. Consumer doesn't consume it. latest offset = 9
5) You start consumer and start stream from offset 9 and lose 8 and 9 messages.